### ISSUE TITLE
filterDeep without the nested path

### ISSUE THREAD

**Issue by kissu** [YuriGor/deepdash#70]
Hi ! 👋🏻 
Thanks you for the awesome package. 🙏🏻 

I'm asking for some help. Especially the `filterDeep` method.

I do currently have an object like this
```json
{
  "a": {
    "b": {
      "c": {
      ...,
      "nestedItem": {
        "matchMe": true,
        "name": "cool name",
        "url": "https://"
      },
      ...,
      }
    }
  }
}
```

and I'm filtering by, let's say `matchMe`. I want to get the property and all the others at the same level (my wished result is not nested).

The issue is that I do have to access that kind of structure: `a.b.c.nestedItem` but I do wish to have only the object
```js
{
  "matchMe": true,
  "name": "cool name",
  "url": "https://"
}
```
Basically what a `findDeep` returns. ☺️

I tried various options like `includeRoot`, `rootIsChildren` etc...but none was successful. 😞

**Comment by YuriGor** [comment]
No feedback, closing for now, let me know if you still have some questions.

**Comment by kissu** [comment]
Hey !
Sorry for the late response (lost 2 days on some damn plumbing 😆 💧 ...). 🙀 
Sorry also for not being precise enough on my question.

So, I basically wanted to have a filter but to have an object or a whole array at the end, **without** the whole path to the object, just the object.
So basically a `filterDeep` + `findDeep` at the same time.

Then I saw #42 and realized that I could simply use `reduceDeep` to achieve what I was looking for.

So, I took my funky JSON and transformed into a JS object
```js
fancyObject: {
  a: {
    amIaFruit: {
      matchMe: true,
      name: 'nice',
      url: 'https://www...',
    },
    b: {
      c: {
        d: {
          matchMe: false,
          name: 'super',
          url: 'https://www...',
        },
      },
    },
    d: {
      g: {
        h: {
          i: {
            j: {
              h: {
                matchMe: true,
                name: 'tasty',
                url: 'https://www...',
                children: {
                  matchMe: true,
                  name: 'santa',
                  url: 'https://www...',
                },
              },
            },
          },
        },
        matchMe: true,
        name: 'penguin',
        url: 'https//www...',
      },
    },
  },
}
```

Then I used the reducer, simply and totally flexible with `parent` if needed !
```js
const filtered = reduceDeep(
  this.fancyObject,
  (acc, value, key, parent) => {
    if (key === 'matchMe' && value) acc.push(parent)
    return acc
  },
  [],
  { leavesOnly: true } // not sure if this is actually needed, can't see difference
)
```

And here we are, a totally clean payload to use on the frontend ! 😃 

```js
[{
  "amIaFruit": {
    "matchMe": true,
    "name": "nice",
    "url": "https://www..."
},
{
  "matchMe": true,
  "name": "tasty",
  "url": "https://www..."
},
{
  "matchMe": true,
  "name": "santa",
  "url": "https://www..."
},
{
  "matchMe": true,
  "name": "penguin",
  "url": "https//www..."
}]
```

Posted this here, if somebody ever needs it someday.
Thanks again for your work Yuri ! 🙏🏻

**Comment by YuriGor** [comment]
Ok, got it, so you actually wanted [selectDeep](https://github.com/YuriGor/deepdash/issues/48) which is not implemented yet :) so yes, your solution is most optimal currently.

**Comment by kissu** [comment]
Yeah, that's what I saw but still managed to do it in a clean way. ❤️
